eureka-client/src/main/java/com/netflix/appinfo/providers/VipAddressResolver.java

```diff
@@ -0,0 +1,15 @@
+package com.netflix.appinfo.providers;
+
+/**
+ * This only really exist for legacy support
+ */
+public interface VipAddressResolver {
+
+    /**
+     * Convert <code>VIPAddress</code> by substituting environment variables if necessary.
+     *
+     * @param vipAddressMacro the macro for which the interpolation needs to be made.
+     * @return a string representing the final <code>VIPAddress</code> after substitution.
+     */
+    String resolveDeploymentContextBasedVipAddresses(String vipAddressMacro);
```

Comment:
I realize this class is temporary but the class name and especially the method name seem extremely specific.  Why not create an interpolator api instead.  It greatly clarifies the purpose of this class.

```java
public interface StringInterpolator {
    String interpolate(String source);
}
```